prog/prio.go

```diff
@@ -235,6 +235,7 @@ func (target *Target) BuildChoiceTable(corpus []*Prog, enabled map[*Syscall]bool
 				sum += prios[i][j]
 			}
 			run[i][j] = sum
+			sum = 0
```

Comment:
I don't think this is correct.
This is specifically a _running_ sum. See below on line 262 we are doing a binary search using a random number 
in the range [0..max]. This is a way to generate weighted random numbers in O(log(N)) time.
We do a similar thing in mutation.go (search for sort.Search).
But you may change this PR to add proper comments to this code to explain to future readers :)

Reply:
@dvyukov ack, I will add comments to the code shortly.
For my better understanding, could you explain why weighted random numbers are used instead of the priority directly? From what I see, the array of runs for a given system call would be strictly increasing and when we perform the search in sort.Search (example https://github.com/google/syzkaller/blob/bcdf85f8bd3fccff5bc9507a589c4847d9b35405/prog/prio.go#L261-L263), there is an implicit bias for the higher values. 
Maybe I'm not understanding it fully.

Reply:
To answer this question I need to ask: how would you use priorities directly? What would be the data structure and the biased random selection?

Reply:
Oh! I understand this approach now! Another way could be to capture these in priority queues but then that would consume more memory. Thanks for the assist Dmitry. I will get a new PR in shortly for adding comments on this. 
I'm currently investigating program generation and needed to consume the priorities in a different manner.